### PR TITLE
Refactor Filtering Updated

### PR DESCRIPTION
**Pull request by vianxnguyen**
## Overview

Refactor filtering updated to take into account of new content type for flyers, replacing #122.

## Changes Made

Refer to #122 

## Test Coverage
Simulator.

## Related PRs or Issues (delete if not applicable)

Replaces #122 

## Screenshots 

https://user-images.githubusercontent.com/46759999/232266336-62892e42-e434-4426-9025-38b1b796b35c.mp4

### CODE REVIEW

File: Volume/Models/Filter.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//
+//  Filter.swift
+//  Volume
+//
+//  Created by Vian Nguyen on 4/15/23.
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
```
Comment (vinnie4k) [inline Volume/Models/Filter.swift:10]: I don't think you need to import SwiftUI here.

File: Volume/Views/Search/SearchResultsList.swift [hunk 2]
```diff
@@ -94,36 +94,38 @@ struct SearchResultsList: View {
...
+                
+                fetchContent(done)
+            } content: {
                 switch selectedTab {
                 case .articles:
                     articleSection
                 case .magazines:
                     magazineSection
+                case .flyers:
+                    SkeletonView()
```
Comment (vinnie4k) [inline Volume/Views/Search/SearchResultsList.swift:119]: Can you add a TODO to implement the articles search section once it's done on the backend?

File: Volume/Views/Search/SearchResultsList.swift [hunk 3]
```diff
@@ -94,36 +94,38 @@ struct SearchResultsList: View {
...
-                sectionStates.articles = .reloading(articles)
-                sectionStates.magazines = .reloading(magazines)
-            }
-
-            sectionStates.articles = .loading
-            sectionStates.magazines = .loading
-
-            fetchContent(done)
-            }) {
+        VStack {
```
Comment (vinnie4k) [inline Volume/Views/Search/SearchResultsList.swift:97]: Is a VStack necessary here? You didn't have it before.

File: Volume/Views/Search/SearchView.swift [hunk 4]
```diff
@@ -22,21 +22,18 @@ struct SearchView: View {
...
                     SearchDropdownView(searchState: $searchState, searchText: $searchText)
+                        .padding([.horizontal, .top])
                 case .results:
                     SearchResultsList(searchText: searchText)
                         .transition(.move(edge: .trailing))
                 }
             }
             .foregroundColor(.black)
-            .padding(EdgeInsets(top: Constants.sectionVerticalPadding, leading: Constants.sectionHorizontalPadding,
-                    bottom: 0, trailing: Constants.sectionHorizontalPadding))
```
Comment (vinnie4k) [inline Volume/Views/Search/SearchView.swift:39]: Can you remove the constants that are not in use anymore?

**Review by vinnie4k — APPROVED** [review]
Looks good! Just a couple of comments. Hopefully my refactoring of the codebase made things a lot more neat.

**Review by ReadeP — APPROVED** [review]

**Pull request merged**